src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -116,7 +116,12 @@ public class Query : IDisposable
 
             // Process the query into batches 
             BatchParserWrapper parser = new BatchParserWrapper();
-            List<BatchDefinition> parserResult = parser.GetBatches(queryText);
+            ExecutionEngineConditions conditions = null;
+            if (settings.IsSqlCmdMode)
+            {
+                conditions = new ExecutionEngineConditions() { IsSqlCmd = settings.IsSqlCmdMode };
+            }
+            List<BatchDefinition> parserResult = parser.GetBatches(queryText, conditions);
```

Comment:
Adding a comment like the following would help with the maintainability:
            if (settings.IsSqlCmdMode)
            {
                conditions = new ExecutionEngineConditions() { IsSqlCmd = settings.IsSqlCmdMode };
            }
             // else  conditions remains null and is handled correctly by parser.GetBatches to continue to nun in non-SqlCmdMode correctly